beacon-chain/db/kv/state.go

```diff
@@ -144,10 +147,13 @@ func (k *Store) DeleteStates(ctx context.Context, blockRoots [][32]byte) error {
 			return err
 		}
 
+		bkt = tx.Bucket(blocksBucket)
+		headBlkRoot := bkt.Get(headBlockRootKey)
+
 		for _, blockRoot := range blockRoots {
-			// Safe guard against deleting genesis or finalized state.
-			if bytes.Equal(blockRoot[:], checkpoint.Root) || bytes.Equal(blockRoot[:], genesisBlockRoot) {
-				return errors.New("could not delete genesis or finalized state")
+			// Safe guard against deleting genesis, finalized, or head state.
+			if bytes.Equal(blockRoot[:], checkpoint.Root) || bytes.Equal(blockRoot[:], genesisBlockRoot) || bytes.Equal(blockRoot[:], headBlkRoot) {
```

Comment:
Duplicated logic here, can it be its own helper func?

Reply:
+1

Reply:
Looking at this again, it's a bit difficult to refactor this. I suppose I could write a method like
```go
func ensureStateRoots(blockRoot, finalizedCheckpointRoot, genesisBlockRoot, headBlockRoot [32]byte) bool {
  return bytes.Equal(blockRoot[:], checkpoint.Root) || bytes.Equal(blockRoot[:], genesisBlockRoot) || bytes.Equal(blockRoot[:], headBlkRoot) 
}
```

I don't know if this adds much value though. What do you think?

Reply:
imo, in that case lets just keep it as it is, if there isnt much benefit to changing it